frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts

```diff
@@ -628,6 +628,32 @@ const interpolateTimeSeriesData = (
   return result;
 };
 
+export function scaleDataset(
+  dataset: ChartDataset,
+  seriesModels: SeriesModel[],
+  settings: ComputedVisualizationSettings,
+): ChartDataset {
+  const transformFn = (datum: Datum) => {
+    const transformedRecord = { ...datum };
+    for (const seriesModel of seriesModels) {
+      const scale = getColumnScaling(seriesModel.column, settings);
+
+      const key = seriesModel.dataKey;
+      if (key in datum) {
+        const scaledValue = Number.isFinite(datum[key])
+          ? (datum[key] as number) * scale
+          : null;
+        transformedRecord[key] = scaledValue;
+      }
+    }
+    return transformedRecord;
+  };
+
+  return dataset.map(datum => {
+    return transformFn(datum);
+  });
+}
+
```

Comment:
We only need to use the scale factor once for transforming the dataset and then we can disregard the fact that it was ever scaled thereon after. Eventually it would nice to rip out the additional scaling in the value formatter, so that we do not have to maintain that across many different contexts.